### ISSUE TITLE
ListTile not using ThemeData primaryColor as default selected color

### ISSUE THREAD

**Issue by nicodumdum** [flutter/flutter#84633]
## Summary of the Issue
I was using the Flutter version `2.0.6` and when I upgraded to version `2.2.2`, there were some discrepancies on the behavior of some of the material widgets. First, I noticed that the `ListTile`s in my app drawer had changed their default selected color from the `ThemeData` `primaryColor` that I had set which was teal. According to the docs: `By default the selected color is the theme's primary color. The selected color can be overridden with a [ListTileTheme].` So the selected color should have been teal instead of blue. Then, I also noticed similar behavior with the `TextFormField`s default focused color.

I hope you can help.

Thanks.

## Steps to Reproduce

1. Run `main.dart`.
2. Tap on the `TextFormField` to focus and see the focus color.
3. Tap on the drawer button to show the drawer and see the `ListTile` selected color.

<details>
  <summary>main.dart</summary>

```
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        brightness: Brightness.light,
        primaryColor: ColorName.teal,
        accentColor: ColorName.tealAccent,
        //! Flutter 2 Regression Issue
        //! Status bar text/icons turned into black/dark instead of white/light.
        //! Setting app bar theme brightness to dark as workaround.
        //! https://github.com/flutter/flutter/issues/72562
        appBarTheme:
            Theme.of(context).appBarTheme.copyWith(brightness: Brightness.dark),
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      drawer: MyDrawer(),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text(
              'Hello World',
            ),
            TextFormField(
              decoration: InputDecoration(
                hintText: 'Email Address',
                prefixIcon: Icon(Icons.alternate_email),
              ),
              keyboardType: TextInputType.emailAddress,
              autofillHints: [AutofillHints.email],
            ),
          ],
        ),
      ),
    );
  }
}

class MyDrawer extends StatefulWidget {
  MyDrawer({Key key}) : super(key: key);

  @override
  MyDrawerState createState() => MyDrawerState();
}

class MyDrawerState extends State<MyDrawer> {
  @override
  void initState() {
    super.initState();
  }

  @override
  void dispose() {
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Drawer(
      child: ListView(
        padding:
            EdgeInsets.zero, // Important: Remove any padding from the ListView.
        children: <Widget>[
          _buildDrawerHeader(),
          _buildDrawerItem(
            icon: Icons.star,
            text: 'Menu Item 1',
            onTap: () {
              Navigator.pop(context);
            },
            isSelected: true,
          ),
          _buildDrawerItem(
            icon: Icons.star,
            text: 'Menu Item 2',
            onTap: () {
              Navigator.pop(context);
            },
            isSelected: false,
          ),
        ],
      ),
    );
  }

  Widget _buildDrawerHeader() {
    return DrawerHeader(
      decoration: BoxDecoration(
        color: Theme.of(context).primaryColor,
      ),
      child: Align(
        alignment: Alignment.bottomLeft,
        child: Text(
          'Main Menu',
          style: Theme.of(context)
              .textTheme
              .headline6
              .copyWith(color: Colors.white),
        ),
      ),
    );
  }

  Widget _buildDrawerItem({
    @required IconData icon,
    @required String text,
    @required GestureTapCallback onTap,
    bool isSelected = false,
  }) {
    return ListTile(
      leading: Icon(icon),
      title: Text(text),
      onTap: onTap,
      selected: isSelected,
    );
  }
}

class ColorName {
  ColorName._();

  static const MaterialColor teal = MaterialColor(
    0xFF115272,
    <int, Color>{
      50: Color(0xFFE2EAEE),
      100: Color(0xFFB8CBD5),
      200: Color(0xFF88A9B9),
      300: Color(0xFF58869C),
      400: Color(0xFF356C87),
      500: Color(0xFF115272),
      600: Color(0xFF0F4B6A),
      700: Color(0xFF0C415F),
      800: Color(0xFF0A3855),
      900: Color(0xFF052842),
    },
  );
  static const MaterialAccentColor tealAccent = MaterialAccentColor(
    0xFF00C7EB,
    <int, Color>{
      100: Color(0xFF29DEFF),
      200: Color(0xFF00C7EB),
      400: Color(0xFF00B1D1),
      700: Color(0xFF00A4C2),
    },
  );
}

```

</details>

**Expected results (Using Flutter 2.0.6):** 
![Screenshot_20210615-205228 (2)](https://user-images.githubusercontent.com/1650987/122067583-45334880-ce26-11eb-8759-a5b3b2c611ae.png)
![Screenshot_20210615-205221](https://user-images.githubusercontent.com/1650987/122067608-4a909300-ce26-11eb-80c0-f9b833f5a012.png)

**Actual results (Using Flutter 2.2.2):** 
![Screenshot_20210615-211051](https://user-images.githubusercontent.com/1650987/122067638-511f0a80-ce26-11eb-8ebb-33c3ea73f189.png)
![Screenshot_20210615-211042](https://user-images.githubusercontent.com/1650987/122067654-53816480-ce26-11eb-95c3-7bbe552166f7.png)

**Comment by TahaTesser** [comment]
Hi @nicodumdum 
For the material widget, primaryColor is changed primary from ColorScheem


You need to use  `primary` in  ColorScheme instead primaryColor`

In this code sample you can see text is purple 


<details> 
<summary>minimal code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(
    MaterialApp(
      theme: ThemeData(
        primaryColor: Colors.red,
        colorScheme: ColorScheme.light().copyWith(
          primary: Colors.purple
        )
      ),
      home: Scaffold(
        body: ListTile(
          title: const Text('ListTile with red background'),
          selected: true,
          
        ),
      ),
    ),
  );
}


``` 

</details>


<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel stable, 2.2.2, on Microsoft Windows [Version 10.0.19043.1055], locale en-US)
    • Flutter version 2.2.2 at C:\Users\Taha\Code\flutter_stable
    • Framework revision d79295af24 (5 days ago), 2021-06-11 08:56:01 -0700
    • Engine revision 91c9fc8fe0
    • Dart version 2.13.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Taha\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_SDK_ROOT = C:\Users\Taha\Code\SDK
    • Java binary at: C:\Users\Taha\Code\openjdk\bin\java.exe
    • Java version OpenJDK Runtime Environment (build 1.8.0-262-b10)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.10.1)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.10.31402.337
    • Windows 10 SDK version 10.0.19041.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[✓] IntelliJ IDEA Community Edition (version 2021.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2021.1.1
    • Flutter plugin version 57.0.5
    • Dart plugin version 211.7233

[✓] VS Code (version 1.57.0)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.23.0

[✓] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.1055]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 91.0.4472.101
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 91.0.864.48

! Doctor found issues in 1 category.
```

</details>

**Comment by nicodumdum** [comment]
Hi @TahaTesser,

Thanks for the response. I realized I should have just used `ThemeData`'s `primarySwatch` with my `MaterialColor` `teal` instead of `primaryColor` and it showed my desired results for both `ListTile` and `TextFormField`. I had also read about the [deprecation](https://flutter.dev/docs/release/breaking-changes/theme-data-accent-properties) of `ThemeData`'s accent properties so I will just ignore my `tealAccent` color for now.

I'm closing this as my issue had been resolved.

Thanks again.
